src/com/palmergames/bukkit/towny/object/WorldCoord.java

```diff
@@ -30,35 +32,52 @@
 
 public class WorldCoord extends Coord {
 
-	private final World world;
+	private final String worldName;
+	private UUID worldUUID;
+	private Reference<World> worldRef = new WeakReference<>(null);
 
 	public WorldCoord(String worldName, int x, int z) {
 		super(x, z);
-		this.world = BukkitTools.getWorld(worldName);
+		this.worldName = worldName;
+		
+		World world = Bukkit.getServer().getWorld(worldName);
+		if (world != null)
+			this.worldUUID = world.getUID();
 	}
 
 	public WorldCoord(String worldName, Coord coord) {
-		super(coord);
-		this.world = BukkitTools.getWorld(worldName);
+		this(worldName, coord.getX(), coord.getZ());
 	}
 
-	public WorldCoord(UUID worldUID, int x, int z) {
+	public WorldCoord(String worldName, UUID worldUUID, int x, int z) {
 		super(x, z);
-		this.world = BukkitTools.getWorld(worldUID);
+		this.worldName = worldName;
+		this.worldUUID = worldUUID;
 	}
 
-	public WorldCoord(UUID worldUID, Coord coord) {

```

Comment:
These should probably get kept. I checked a couple plugins to see if I'm using either of these constructors and couldn't find any thing, still they might be used somewhere.

Post-uuid branch merge, they might be fairly crucial as well. TownyCamps and TownyHistories are saving WCs using the world name for instance, which should probably get changed.

We would have to do the opposite of what we're doing in the `WorldCoord(String worldName, int x, int z)` constructor, getting the name from the UUID, so there's a valid worldName present.

Reply:
I wanted to avoid having a constructor that doesn't have a worldName param, so that the worldcoord at least always has a worldname even if the world isn't loaded in bukkit.